improve image tests

dorschw opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46177

## Description
- Added a retry mechanism to the validation that checks remote images.
- Only checking for http status code without downloading the image checked (thanks @ilaner for the idea)

## Must have
- [x] Tests

DeanArbel reviewed (approved):
Impressive changes. Well done.

tkatzir commented on demisto_sdk/commands/common/errors.py:
```diff
@@ -1646,14 +1646,19 @@ def invalid_readme_insert_image_link_error(path):
+    def invalid_readme_image_error(path: str, error_type: str, http_code: Optional[int] = None):
+        error = 'Error in readme image: '
+        if http_code:
+            error += f'got HTTP response code {http_code}'
```
Need to add an extra space at the end?

tkatzir replied:
Print the `reason` param as well

tkatzir commented on demisto_sdk/commands/common/errors.py:
```diff
@@ -1646,14 +1646,19 @@ def invalid_readme_insert_image_link_error(path):
-            'branch_name_readme_absolute_error': Errors.branch_name_in_readme_image_absolute_path_error,
-            'insert_image_link_error': Errors.invalid_readme_insert_image_link_error
-        }.get(error_type, lambda x: f'Something went wrong when testing {x}')(path)
+    def invalid_readme_image_error(path: str, error_type: str, http_code: Optional[int] = None):
```
Accept the `reason` param as well

tkatzir reviewed (changes requested):
Great work!
Missing an extra space at the end, for readability.

idovandijk commented:
@dorschw can u please explain to me in short what this change is?

dorschw commented:
> @dorschw can u please explain to me in short what this change is?

Updated PR description:
```
## Description
- Added a retry mechanism to the validation that checks remote images.
- Only checking for http status code without downloading the image checked (thanks @ilaner for the idea)
```

idovandijk reviewed (approved):
So nothing for me to review here, if they approved it's fine
